File: src/lib-common/slice.js
function slice(x,y,z)
{
 check(is_vec,x)
 
 const n1=inc(x.length)
 
 check(between,y,0,n1)
 
 if(is_undef(z))
 {
  const n=sub(x.length,y)
  
  return slice(x,y,n)  
 }

 check(between,z,0,n1)
 
 const n2=add(y,z)

 check(between,n2,0,n1)
 
 return x.slice(y,n2)
}

File: src/lib-common/slice.cs
fn slice x:vec index:uint length
 check lte index x.length

 if is_undef length
  let n sub x.length index

  ret slice x index n
 end

 check is_uint length
 check lte length x.length
 check lte index x.length

 let n add index length

 check lte n x.length

 let r x.slice index n

 check same r.length length

 ret r
end
